src/HIE/Bios/Environment.hs

```diff
@@ -120,10 +120,10 @@ setHiDir f d = d { hiDir      = Just f}
 -- It would be good to move this code into a library module so we can just use it
 -- rather than copy it.
 addCmdOpts :: (GhcMonad m)
-           => [String] -> DynFlags -> m (DynFlags, [G.Target])
-addCmdOpts cmdOpts df1 = do
-  (df2, leftovers', _warns) <- G.parseDynamicFlags df1 (map G.noLoc cmdOpts)
-
+           => FilePath -> [String] -> DynFlags -> m (DynFlags, [G.Target])
```

Comment:
Can you not modify `addCmdOpts` and instead write a different function which manipulates the `DynFlags` after they are parsed? It is fine to want to use `addCmdOpts ` without knowing the root directory.